boxoffice/models/discount_policy.py

```diff
@@ -0,0 +1,75 @@
+from boxoffice.models import db, BaseScopedNameMixin, IdMixin
+from datetime import datetime
+from boxoffice.models import ItemCollection
+from baseframe import __
+from coaster.utils import LabeledEnum
+
+import string
+import random
+
+__all__ = ['DiscountPolicy', 'DiscountCoupon', 'item_discount_policy']
+
+
+def generate_coupon_code(size=6, chars=string.ascii_uppercase + string.digits):
+    return ''.join(random.choice(chars) for _ in range(size))
+
+
+class DISCOUNTTYPES(LabeledEnum):
+    AUTOMATIC = (0, __("Automatic"))
+    COUPON = (1, __("Coupon"))
+
+
+item_discount_policy = db.Table('item_discount_policy', db.Model.metadata,
+    db.Column('item_id', None, db.ForeignKey('item.id'), primary_key=True),
+    db.Column('discount_policy_id', None, db.ForeignKey('discount_policy.id'), primary_key=True),
+    db.Column('created_at', db.DateTime, default=datetime.utcnow, nullable=False))
+
+
+class DiscountPolicy(BaseScopedNameMixin, db.Model):
+    """
+    Consists of discount rules for prices applied on items
+    """
+    __tablename__ = 'discount_policy'
+    __uuid_primary_key__ = True
+    __table_args__ = (db.UniqueConstraint('item_collection_id', 'name'),
+    db.CheckConstraint('item_quantity_min <= item_quantity_max', 'item_quantity_range'),
+    db.CheckConstraint('percentage <= 100', 'percentage_bound_upper'),
+    db.CheckConstraint('percentage > 0', 'percentage_bound_lower'))
```

Comment:
1. All three lines should be indented
2. Percentage checks can be a single check constraint using SQL `and`
3. Let's come up with a consistent naming policy for check constraints, like `table_column_check` or `check_table_column` (the names are at db-level not table-level, so `percentage_bound_upper` will potentially clash with another)